### PR TITLE
Initial commit from merging summit demo

### LINKED ISSUE #31: Migrate Jenkins to use Provided Image/Configuration
Migrate the Jenkins server from a custom built instance to instead leverage the image shipped from OpenShift

### PR DESCRIPTION
**Pull request by sabre1041**
Merges the changes from the Summit demo. 

Be sure to follow steps to create additional PV's as per README

Resolves #31 #29 #28 #25 #24 #34

### CODE REVIEW

File: projects/coolstore-kjar-s2i/Jenkinsfile [hunk 1]
```diff
@@ -106,9 +114,90 @@ node {
...
+   full_version=\$(cat $sourceFileName)
+   parsed_version=\${full_version:33}
+   mkdir -p \$(dirname ${destinationFileName})
+   echo \$parsed_version > ${destinationFileName}
+    """
+    
+    return readFile("${destinationFileName}").trim()
+}
+
+def parseImageVersion(String sourceFileName, String destinationFileName, String rulesVersion, String buildVersion) {
```
Comment (jgoldsmith613) [inline projects/coolstore-kjar-s2i/Jenkinsfile]: Not using sourceFileName

File: projects/coolstore-kjar-s2i/Jenkinsfile [hunk 2]
```diff
@@ -1,42 +1,46 @@
...
+    def devProject = "coolstore-bdd-dev"
+    def prodProject = "coolstore-bdd-prod"
    
    stage 'Checkout'
 
        checkout scm
        
   stage 'Retrieve Rules'
 
+      def rulesVersion = parseVersion("${WORKSPACE}/version.properties", "${WORKSPACE}/target/parsed_version.txt")
```
Comment (jgoldsmith613) [inline projects/coolstore-kjar-s2i/Jenkinsfile:15]: Never using destination file

File: projects/coolstore-kjar-s2i/Jenkinsfile (hunk 1 above)
Comment (sabre1041) [inline projects/coolstore-kjar-s2i/Jenkinsfile]: I battled with this method for 6hrs and it went through a bunch of refactoring. we can remove that parameter

**Comment by sabre1041** [comment]
@jgoldsmith613 I updated the PR and removed the _sourceFileName_ variable

**Comment by sabre1041** [comment]
@jgoldsmith613 Added support for nodejs. Please review

**Pull request merged**